addons/token-exchange/agent_mirrorpeer_controller.go

```diff
@@ -225,42 +227,42 @@ func (r *MirrorPeerReconciler) createVolumeReplicationClass(ctx context.Context,
 		klog.Error(err, "Failed to get current storage cluster ref")
 		return err
 	}
-	params := make(map[string]string)
-	params[MirroringModeKey] = string(mp.Spec.Mode)
-	params[SchedulingIntervalKey] = mp.Spec.SchedulingInterval
-	params[ReplicationSecretNameKey] = mp.Spec.ReplicationSecretName
-	params[ReplicationSecretNamespaceKey] = scr.Namespace
-	vrc := &replicationv1alpha1.VolumeReplicationClass{
-		ObjectMeta: metav1.ObjectMeta{
-			Name: "odf-rbd-volumereplicationclass",
-		},
-		Spec: replicationv1alpha1.VolumeReplicationClassSpec{
+
+	for _, interval := range mp.Spec.SchedulingInterval {
+		params := make(map[string]string)
+		params[MirroringModeKey] = DefaultMirroringMode
+		params[SchedulingIntervalKey] = interval
+		params[ReplicationSecretNameKey] = ReplicationSecretName
+		params[ReplicationSecretNamespaceKey] = scr.Namespace
+		vrcName := fmt.Sprintf(VolumeReplicationClassNameTemplate, common.FnvHash(interval))
```

Comment:
why do we need to hash this? for same interval value, it will always collide. Instead let's directly append interval?

Reply:
@Madhu-1 If two intervals are same, do they point towards same VRC or is it a one to one mapping ?

Reply:
@umangachapagain I prefer using  hashed value for naming the VRC instead of appending the interval. This will avoid the users from creating their own VRCs and abstract  our logic

Reply:
Users can create their own VRC.
No harm in keeping it I guess.

Reply:
its not useful to have 2 VR with same intervals.